File: 0153_Find_Minimum_in_Rotated_Sorted_Array/my_ans.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0;
        int right = nums.size()-1;
        
        while (left < right) {
            int medium = (left + right) / 2;
            if (nums[left] < nums[right]) return nums[left];
            if (nums[right] < nums[medium]) {
                left = medium + 1;
            } else {
                right = medium;
            }

        }
        return nums[left];
    }
};

File: 0035_Search_Insert_Position/myAns.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let nums_size = nums.len()-1;
        let mut left = 0;
        let mut right = nums_size;

        while left <= right {
            let medium = left + (right-left)/2;
            if (nums[medium] == target) {
                return medium as i32;
            } else if (nums[medium] < target) {
                left = medium + 1;
            } else {
                if medium == 0 {
                    break;
                }
                right = medium - 1;
            }
        }

        left as i32
    }
}